static/scripts/rewards/gift-cards/index.ts

```diff
@@ -0,0 +1,85 @@
+import { allowedCountries } from "../../../../shared/allowed-country-list";
+import { getGiftCardOrderId, isGiftCardAvailable } from "../../../../shared/helpers";
+import { GiftCard, OrderTransaction } from "../../../../shared/types";
+import { AppState } from "../app-state";
+import { getGiftCardHtml } from "./gift-card";
+import { getApiBaseUrl, getUserCountryCode } from "./helpers";
+import { attachMintAction } from "./mint/mint-action";
+import { getRedeemCodeHtml } from "./reveal/redeem-code-html";
+import { attachRevealAction } from "./reveal/reveal-action";
+
+export async function initClaimGiftCard(app: AppState) {
+  const giftCardsSection = document.getElementById("gift-cards");
+  if (!giftCardsSection) {
+    console.error("Missing gift cards section #gift-cards");
+    return;
+  }
+  giftCardsSection.innerHTML = "Loading...";
+
+  const country = await getUserCountryCode();
+  if (!country) {
+    giftCardsSection.innerHTML = `<p class="card-error">Failed to load suitable virtual cards for you. Refresh or try disabling adblocker.</p>`;
+    return;
+  }
+
+  if (!allowedCountries.find((allowedCountry) => allowedCountry.code == country)) {
+    giftCardsSection.innerHTML = `<p class="card-error">Virtual cards are not available for your location. Use other methods to claim your reward.</p>`;
+    return;
+  }
+
+  const retrieveOrderUrl = `${getApiBaseUrl()}/get-order?orderId=${getGiftCardOrderId(app.reward.beneficiary, app.reward.signature)}`;
+  const bestCardUrl = `${getApiBaseUrl()}/get-best-card?country=${country}&amount=${app.reward.amount}`;
+
+  const requestInit = {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+    },
+  };
+
+  const [orderResponse, bestCardResponse] = await Promise.all([fetch(retrieveOrderUrl, requestInit), fetch(bestCardUrl, requestInit)]);
+
+  const giftCard = (await bestCardResponse.json()) as GiftCard;
+
+  if (orderResponse.status == 200) {
+    const { transaction, product } = (await orderResponse.json()) as {
+      transaction: OrderTransaction;
+      product: GiftCard | null;
```

Comment:
product is nullable here following a successful order response, how can that be?

Reply:
This is the frontend side of your [above](https://github.com/ubiquity/pay.ubq.fi/pull/277#discussion_r1743543545) review comment. Reason is same for product being null in frontend or backend. I have given details there. So, let's discuss it there.